x-pack/plugins/enterprise_search/public/applications/workplace_search/views/content_sources/components/add_source/github_via_app_logic.ts

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { kea, MakeLogicType } from 'kea';
+
+import { flashAPIErrors, flashSuccessToast } from '../../../../../shared/flash_messages';
+import { HttpLogic } from '../../../../../shared/http';
+import { KibanaLogic } from '../../../../../shared/kibana';
+import { AppLogic } from '../../../../app_logic';
+import { SOURCES_PATH, getSourcesPath } from '../../../../routes';
+import { ContentSourceFullData } from '../../../../types';
+
+interface GithubViaAppValues {
+  githubAppId: string;
+  githubEnterpriseServerUrl: string;
+  stagedPrivateKey: string | null;
+  isSubmitButtonLoading: boolean;
+  indexPermissionsValue: boolean;
+}
+
+interface GithubViaAppActions {
+  setGithubAppId(githubAppId: string): string;
+  setGithubEnterpriseServerUrl(githubEnterpriseServerUrl: string): string;
+  setStagedPrivateKey(stagedPrivateKey: string | null): string | null;
+  setButtonNotLoading(): void;
+  createContentSource(): void;
+  setSourceIndexPermissionsValue(indexPermissionsValue: boolean): boolean;
+}
+
+export const GithubViaAppLogic = kea<MakeLogicType<GithubViaAppValues, GithubViaAppActions>>({
+  path: ['enterprise_search', 'workplace_search', 'github_via_app_logic'],
+  actions: {
+    setGithubAppId: (githubAppId: string) => githubAppId,
+    setGithubEnterpriseServerUrl: (githubEnterpriseServerUrl: string) => githubEnterpriseServerUrl,
+    createContentSource: () => true,
+    setStagedPrivateKey: (stagedPrivateKey: string) => stagedPrivateKey,
+    setButtonNotLoading: () => false,
```

Comment:
Slightly more performant to make this a property and not a method.
```suggestion
    setButtonNotLoading: false,
```